// googleapis/build/src/index.js
"use strict";
/**
 * Copyright 2017 Google Inc. All Rights Reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
Object.defineProperty(exports, "__esModule", { value: true });
/*! THIS FILE IS AUTO-GENERATED */
const googleapis_1 = require("./googleapis");
exports.GoogleApis = googleapis_1.GoogleApis;
const google = new googleapis_1.GoogleApis();
exports.google = google;
var v1_1 = require("./apis/abusiveexperiencereport/v1");
exports.abusiveexperiencereport_v1 = v1_1.abusiveexperiencereport_v1;
var v1_2 = require("./apis/acceleratedmobilepageurl/v1");
exports.acceleratedmobilepageurl_v1 = v1_2.acceleratedmobilepageurl_v1;
var v1_2_1 = require("./apis/adexchangebuyer/v1.2");
exports.adexchangebuyer_v1_2 = v1_2_1.adexchangebuyer_v1_2;
var v1_3_1 = require("./apis/adexchangebuyer/v1.3");
exports.adexchangebuyer_v1_3 = v1_3_1.adexchangebuyer_v1_3;
var v1_4_1 = require("./apis/adexchangebuyer/v1.4");
exports.adexchangebuyer_v1_4 = v1_4_1.adexchangebuyer_v1_4;
var v2beta1_1 = require("./apis/adexchangebuyer2/v2beta1");
exports.adexchangebuyer2_v2beta1 = v2beta1_1.adexchangebuyer2_v2beta1;
var v1_1_1 = require("./apis/adexchangeseller/v1.1");
exports.adexchangeseller_v1_1 = v1_1_1.adexchangeseller_v1_1;
var v1_3 = require("./apis/adexchangeseller/v1");
exports.adexchangeseller_v1 = v1_3.adexchangeseller_v1;
var v2_0_1 = require("./apis/adexchangeseller/v2.0");
exports.adexchangeseller_v2_0 = v2_0_1.adexchangeseller_v2_0;
var v1_4 = require("./apis/adexperiencereport/v1");
exports.adexperiencereport_v1 = v1_4.adexperiencereport_v1;
var datatransfer_v1_1 = require("./apis/admin/datatransfer_v1");
exports.admin_datatransfer_v1 = datatransfer_v1_1.admin_datatransfer_v1;
var directory_v1_1 = require("./apis/admin/directory_v1");
exports.admin_directory_v1 = directory_v1_1.admin_directory_v1;
var reports_v1_1 = require("./apis/admin/reports_v1");
exports.admin_reports_v1 = reports_v1_1.admin_reports_v1;
var v1_4_2 = require("./apis/adsense/v1.4");
exports.adsense_v1_4 = v1_4_2.adsense_v1_4;
var v4_1_1 = require("./apis/adsensehost/v4.1");
exports.adsensehost_v4_1 = v4_1_1.adsensehost_v4_1;
var v2_4_1 = require("./apis/analytics/v2.4");
exports.analytics_v2_4 = v2_4_1.analytics_v2_4;
var v3_1 = require("./apis/analytics/v3");
exports.analytics_v3 = v3_1.analytics_v3;
var v4_1 = require("./apis/analyticsreporting/v4");
exports.analyticsreporting_v4 = v4_1.analyticsreporting_v4;
var v1_5 = require("./apis/androiddeviceprovisioning/v1");
exports.androiddeviceprovisioning_v1 = v1_5.androiddeviceprovisioning_v1;
var v1_6 = require("./apis/androidenterprise/v1");
exports.androidenterprise_v1 = v1_6.androidenterprise_v1;
var v1_7 = require("./apis/androidmanagement/v1");
exports.androidmanagement_v1 = v1_7.androidmanagement_v1;
var v1_1_2 = require("./apis/androidpublisher/v1.1");
exports.androidpublisher_v1_1 = v1_1_2.androidpublisher_v1_1;
var v1_8 = require("./apis/androidpublisher/v1");
exports.androidpublisher_v1 = v1_8.androidpublisher_v1;
var v2_1 = require("./apis/androidpublisher/v2");
exports.androidpublisher_v2 = v2_1.androidpublisher_v2;
var v3_2 = require("./apis/androidpublisher/v3");
exports.androidpublisher_v3 = v3_2.androidpublisher_v3;
var v1_9 = require("./apis/appengine/v1");
exports.appengine_v1 = v1_9.appengine_v1;
var v1alpha_1 = require("./apis/appengine/v1alpha");
exports.appengine_v1alpha = v1alpha_1.appengine_v1alpha;
var v1beta_1 = require("./apis/appengine/v1beta");
exports.appengine_v1beta = v1beta_1.appengine_v1beta;
var v1beta4_1 = require("./apis/appengine/v1beta4");
exports.appengine_v1beta4 = v1beta4_1.appengine_v1beta4;
var v1beta5_1 = require("./apis/appengine/v1beta5");
exports.appengine_v1beta5 = v1beta5_1.appengine_v1beta5;
var v1_10 = require("./apis/appsactivity/v1");
exports.appsactivity_v1 = v1_10.appsactivity_v1;
var v1_11 = require("./apis/appstate/v1");
exports.appstate_v1 = v1_11.appstate_v1;
var v2_2 = require("./apis/bigquery/v2");
exports.bigquery_v2 = v2_2.bigquery_v2;
var v1_12 = require("./apis/bigquerydatatransfer/v1");
exports.bigquerydatatransfer_v1 = v1_12.bigquerydatatransfer_v1;
var v2_3 = require("./apis/blogger/v2");
exports.blogger_v2 = v2_3.blogger_v2;
var v3_3 = require("./apis/blogger/v3");
exports.blogger_v3 = v3_3.blogger_v3;
var v1_13 = require("./apis/books/v1");
exports.books_v1 = v1_13.books_v1;
var v3_4 = require("./apis/calendar/v3");
exports.calendar_v3 = v3_4.calendar_v3;
var v1_14 = require("./apis/chat/v1");
exports.chat_v1 = v1_14.chat_v1;
var v2_4 = require("./apis/civicinfo/v2");
exports.civicinfo_v2 = v2_4.civicinfo_v2;
var v1_15 = require("./apis/classroom/v1");
exports.classroom_v1 = v1_15.classroom_v1;
var v1_16 = require("./apis/cloudbilling/v1");
exports.cloudbilling_v1 = v1_16.cloudbilling_v1;
var v1_17 = require("./apis/cloudbuild/v1");
exports.cloudbuild_v1 = v1_17.cloudbuild_v1;
var v2_5 = require("./apis/clouddebugger/v2");
exports.clouddebugger_v2 = v2_5.clouddebugger_v2;
var v1beta1_1 = require("./apis/clouderrorreporting/v1beta1");
exports.clouderrorreporting_v1beta1 = v1beta1_1.clouderrorreporting_v1beta1;
var v1_18 = require("./apis/cloudfunctions/v1");
exports.cloudfunctions_v1 = v1_18.cloudfunctions_v1;
var v1beta2_1 = require("./apis/cloudfunctions/v1beta2");
exports.cloudfunctions_v1beta2 = v1beta2_1.cloudfunctions_v1beta2;
var v1_19 = require("./apis/cloudiot/v1");
exports.cloudiot_v1 = v1_19.cloudiot_v1;
var v1_20 = require("./apis/cloudkms/v1");
exports.cloudkms_v1 = v1_20.cloudkms_v1;
var v1_21 = require("./apis/cloudresourcemanager/v1");
exports.cloudresourcemanager_v1 = v1_21.cloudresourcemanager_v1;
var v1beta1_2 = require("./apis/cloudresourcemanager/v1beta1");
exports.cloudresourcemanager_v1beta1 = v1beta1_2.cloudresourcemanager_v1beta1;
var v2_6 = require("./apis/cloudresourcemanager/v2");
exports.cloudresourcemanager_v2 = v2_6.cloudresourcemanager_v2;
var v2beta1_2 = require("./apis/cloudresourcemanager/v2beta1");
exports.cloudresourcemanager_v2beta1 = v2beta1_2.cloudresourcemanager_v2beta1;
var v1_22 = require("./apis/cloudshell/v1");
exports.cloudshell_v1 = v1_22.cloudshell_v1;
var v1alpha1_1 = require("./apis/cloudshell/v1alpha1");
exports.cloudshell_v1alpha1 = v1alpha1_1.cloudshell_v1alpha1;
var v2beta2_1 = require("./apis/cloudtasks/v2beta2");
exports.cloudtasks_v2beta2 = v2beta2_1.cloudtasks_v2beta2;
var v1_23 = require("./apis/cloudtrace/v1");
exports.cloudtrace_v1 = v1_23.cloudtrace_v1;
var v2_7 = require("./apis/cloudtrace/v2");
exports.cloudtrace_v2 = v2_7.cloudtrace_v2;
var v1beta1_3 = require("./apis/composer/v1beta1");
exports.composer_v1beta1 = v1beta1_3.composer_v1beta1;
var alpha_1 = require("./apis/compute/alpha");
exports.compute_alpha = alpha_1.compute_alpha;
var beta_1 = require("./apis/compute/beta");
exports.compute_beta = beta_1.compute_beta;
var v1_24 = require("./apis/compute/v1");
exports.compute_v1 = v1_24.compute_v1;
var v1_25 = require("./apis/container/v1");
exports.container_v1 = v1_25.container_v1;
var v1beta1_4 = require("./apis/container/v1beta1");
exports.container_v1beta1 = v1beta1_4.container_v1beta1;
var v2_8 = require("./apis/content/v2");
exports.content_v2 = v2_8.content_v2;
var v2sandbox_1 = require("./apis/content/v2sandbox");
exports.content_v2sandbox = v2sandbox_1.content_v2sandbox;
var v1_26 = require("./apis/customsearch/v1");
exports.customsearch_v1 = v1_26.customsearch_v1;
var v1b3_1 = require("./apis/dataflow/v1b3");
exports.dataflow_v1b3 = v1b3_1.dataflow_v1b3;
var v1_27 = require("./apis/dataproc/v1");
exports.dataproc_v1 = v1_27.dataproc_v1;
var v1beta2_2 = require("./apis/dataproc/v1beta2");
exports.dataproc_v1beta2 = v1beta2_2.dataproc_v1beta2;
var v1_28 = require("./apis/datastore/v1");
exports.datastore_v1 = v1_28.datastore_v1;
var v1beta1_5 = require("./apis/datastore/v1beta1");
exports.datastore_v1beta1 = v1beta1_5.datastore_v1beta1;
var v1beta3_1 = require("./apis/datastore/v1beta3");
exports.datastore_v1beta3 = v1beta3_1.datastore_v1beta3;
var alpha_2 = require("./apis/deploymentmanager/alpha");
exports.deploymentmanager_alpha = alpha_2.deploymentmanager_alpha;
var v2_9 = require("./apis/deploymentmanager/v2");
exports.deploymentmanager_v2 = v2_9.deploymentmanager_v2;
var v2beta_1 = require("./apis/deploymentmanager/v2beta");
exports.deploymentmanager_v2beta = v2beta_1.deploymentmanager_v2beta;
var v2_8_1 = require("./apis/dfareporting/v2.8");
exports.dfareporting_v2_8 = v2_8_1.dfareporting_v2_8;
var v3_0_1 = require("./apis/dfareporting/v3.0");
exports.dfareporting_v3_0 = v3_0_1.dfareporting_v3_0;
var v3_1_1 = require("./apis/dfareporting/v3.1");
exports.dfareporting_v3_1 = v3_1_1.dfareporting_v3_1;
var v2_10 = require("./apis/dialogflow/v2");
exports.dialogflow_v2 = v2_10.dialogflow_v2;
var v2beta1_3 = require("./apis/dialogflow/v2beta1");
exports.dialogflow_v2beta1 = v2beta1_3.dialogflow_v2beta1;
var v1_29 = require("./apis/digitalassetlinks/v1");
exports.digitalassetlinks_v1 = v1_29.digitalassetlinks_v1;
var v1_30 = require("./apis/discovery/v1");
exports.discovery_v1 = v1_30.discovery_v1;
var v2_11 = require("./apis/dlp/v2");
exports.dlp_v2 = v2_11.dlp_v2;
var v1_31 = require("./apis/dns/v1");
exports.dns_v1 = v1_31.dns_v1;
var v1beta2_3 = require("./apis/dns/v1beta2");
exports.dns_v1beta2 = v1beta2_3.dns_v1beta2;
var v2beta1_4 = require("./apis/dns/v2beta1");
exports.dns_v2beta1 = v2beta1_4.dns_v2beta1;
var v1_32 = require("./apis/doubleclickbidmanager/v1");
exports.doubleclickbidmanager_v1 = v1_32.doubleclickbidmanager_v1;
var v2_12 = require("./apis/doubleclicksearch/v2");
exports.doubleclicksearch_v2 = v2_12.doubleclicksearch_v2;
var v2_13 = require("./apis/drive/v2");
exports.drive_v2 = v2_13.drive_v2;
var v3_5 = require("./apis/drive/v3");
exports.drive_v3 = v3_5.drive_v3;
var v1_33 = require("./apis/firebasedynamiclinks/v1");
exports.firebasedynamiclinks_v1 = v1_33.firebasedynamiclinks_v1;
var v1_34 = require("./apis/firebaserules/v1");
exports.firebaserules_v1 = v1_34.firebaserules_v1;
var v1beta1_6 = require("./apis/firestore/v1beta1");
exports.firestore_v1beta1 = v1beta1_6.firestore_v1beta1;
var v1_35 = require("./apis/fitness/v1");
exports.fitness_v1 = v1_35.fitness_v1;
var v1_36 = require("./apis/fusiontables/v1");
exports.fusiontables_v1 = v1_36.fusiontables_v1;
var v2_14 = require("./apis/fusiontables/v2");
exports.fusiontables_v2 = v2_14.fusiontables_v2;
var v1_37 = require("./apis/games/v1");
exports.games_v1 = v1_37.games_v1;
var v1configuration_1 = require("./apis/gamesConfiguration/v1configuration");
exports.gamesConfiguration_v1configuration = v1configuration_1.gamesConfiguration_v1configuration;
var v1management_1 = require("./apis/gamesManagement/v1management");
exports.gamesManagement_v1management = v1management_1.gamesManagement_v1management;
var v1_38 = require("./apis/genomics/v1");
exports.genomics_v1 = v1_38.genomics_v1;
var v1alpha2_1 = require("./apis/genomics/v1alpha2");
exports.genomics_v1alpha2 = v1alpha2_1.genomics_v1alpha2;
var v2alpha1_1 = require("./apis/genomics/v2alpha1");
exports.genomics_v2alpha1 = v2alpha1_1.genomics_v2alpha1;
var v1_39 = require("./apis/gmail/v1");
exports.gmail_v1 = v1_39.gmail_v1;
var v1_40 = require("./apis/groupsmigration/v1");
exports.groupsmigration_v1 = v1_40.groupsmigration_v1;
var v1_41 = require("./apis/groupssettings/v1");
exports.groupssettings_v1 = v1_41.groupssettings_v1;
var v1_42 = require("./apis/iam/v1");
exports.iam_v1 = v1_42.iam_v1;
var v3_6 = require("./apis/identitytoolkit/v3");
exports.identitytoolkit_v3 = v3_6.identitytoolkit_v3;
var v2_15 = require("./apis/jobs/v2");
exports.jobs_v2 = v2_15.jobs_v2;
var v1_43 = require("./apis/kgsearch/v1");
exports.kgsearch_v1 = v1_43.kgsearch_v1;
var v1_44 = require("./apis/language/v1");
exports.language_v1 = v1_44.language_v1;
var v1beta1_7 = require("./apis/language/v1beta1");
exports.language_v1beta1 = v1beta1_7.language_v1beta1;
var v1beta2_4 = require("./apis/language/v1beta2");
exports.language_v1beta2 = v1beta2_4.language_v1beta2;
var v1_45 = require("./apis/licensing/v1");
exports.licensing_v1 = v1_45.licensing_v1;
var v2_16 = require("./apis/logging/v2");
exports.logging_v2 = v2_16.logging_v2;
var v2beta1_5 = require("./apis/logging/v2beta1");
exports.logging_v2beta1 = v2beta1_5.logging_v2beta1;
var v1_46 = require("./apis/manufacturers/v1");
exports.manufacturers_v1 = v1_46.manufacturers_v1;
var v1_47 = require("./apis/mirror/v1");
exports.mirror_v1 = v1_47.mirror_v1;
var v1_48 = require("./apis/ml/v1");
exports.ml_v1 = v1_48.ml_v1;
var v3_7 = require("./apis/monitoring/v3");
exports.monitoring_v3 = v3_7.monitoring_v3;
var v1_49 = require("./apis/oauth2/v1");
exports.oauth2_v1 = v1_49.oauth2_v1;
var v2_17 = require("./apis/oauth2/v2");
exports.oauth2_v2 = v2_17.oauth2_v2;
var v1_50 = require("./apis/oslogin/v1");
exports.oslogin_v1 = v1_50.oslogin_v1;
var v1alpha_2 = require("./apis/oslogin/v1alpha");
exports.oslogin_v1alpha = v1alpha_2.oslogin_v1alpha;
var v1beta_2 = require("./apis/oslogin/v1beta");
exports.oslogin_v1beta = v1beta_2.oslogin_v1beta;
var v1_51 = require("./apis/pagespeedonline/v1");
exports.pagespeedonline_v1 = v1_51.pagespeedonline_v1;
var v2_18 = require("./apis/pagespeedonline/v2");
exports.pagespeedonline_v2 = v2_18.pagespeedonline_v2;
var v4_2 = require("./apis/pagespeedonline/v4");
exports.pagespeedonline_v4 = v4_2.pagespeedonline_v4;
var v2_19 = require("./apis/partners/v2");
exports.partners_v2 = v2_19.partners_v2;
var v1_52 = require("./apis/people/v1");
exports.people_v1 = v1_52.people_v1;
var v1_53 = require("./apis/playcustomapp/v1");
exports.playcustomapp_v1 = v1_53.playcustomapp_v1;
var v1_54 = require("./apis/plus/v1");
exports.plus_v1 = v1_54.plus_v1;
var v1_55 = require("./apis/plusDomains/v1");
exports.plusDomains_v1 = v1_55.plusDomains_v1;
var v1_56 = require("./apis/poly/v1");
exports.poly_v1 = v1_56.poly_v1;
var v1beta1_8 = require("./apis/proximitybeacon/v1beta1");
exports.proximitybeacon_v1beta1 = v1beta1_8.proximitybeacon_v1beta1;
var v1_57 = require("./apis/pubsub/v1");
exports.pubsub_v1 = v1_57.pubsub_v1;
var v1beta1a_1 = require("./apis/pubsub/v1beta1a");
exports.pubsub_v1beta1a = v1beta1a_1.pubsub_v1beta1a;
var v1beta2_5 = require("./apis/pubsub/v1beta2");
exports.pubsub_v1beta2 = v1beta2_5.pubsub_v1beta2;
var v1beta1_9 = require("./apis/redis/v1beta1");
exports.redis_v1beta1 = v1beta1_9.redis_v1beta1;
var v1beta1_10 = require("./apis/replicapool/v1beta1");
exports.replicapool_v1beta1 = v1beta1_10.replicapool_v1beta1;
var v1beta2_6 = require("./apis/replicapool/v1beta2");
exports.replicapool_v1beta2 = v1beta2_6.replicapool_v1beta2;
var v1beta1_11 = require("./apis/replicapoolupdater/v1beta1");
exports.replicapoolupdater_v1beta1 = v1beta1_11.replicapoolupdater_v1beta1;
var v1_58 = require("./apis/reseller/v1");
exports.reseller_v1 = v1_58.reseller_v1;
var v1_59 = require("./apis/runtimeconfig/v1");
exports.runtimeconfig_v1 = v1_59.runtimeconfig_v1;
var v1beta1_12 = require("./apis/runtimeconfig/v1beta1");
exports.runtimeconfig_v1beta1 = v1beta1_12.runtimeconfig_v1beta1;
var v4_3 = require("./apis/safebrowsing/v4");
exports.safebrowsing_v4 = v4_3.safebrowsing_v4;
var v1_60 = require("./apis/script/v1");
exports.script_v1 = v1_60.script_v1;
var v1_61 = require("./apis/searchconsole/v1");
exports.searchconsole_v1 = v1_61.searchconsole_v1;
var v1_62 = require("./apis/servicebroker/v1");
exports.servicebroker_v1 = v1_62.servicebroker_v1;
var v1alpha1_2 = require("./apis/servicebroker/v1alpha1");
exports.servicebroker_v1alpha1 = v1alpha1_2.servicebroker_v1alpha1;
var v1beta1_13 = require("./apis/servicebroker/v1beta1");
exports.servicebroker_v1beta1 = v1beta1_13.servicebroker_v1beta1;
var v1_63 = require("./apis/serviceconsumermanagement/v1");
exports.serviceconsumermanagement_v1 = v1_63.serviceconsumermanagement_v1;
var v1_64 = require("./apis/servicecontrol/v1");
exports.servicecontrol_v1 = v1_64.servicecontrol_v1;
var v1_65 = require("./apis/servicemanagement/v1");
exports.servicemanagement_v1 = v1_65.servicemanagement_v1;
var v1beta1_14 = require("./apis/serviceusage/v1beta1");
exports.serviceusage_v1beta1 = v1beta1_14.serviceusage_v1beta1;
var v1_66 = require("./apis/serviceuser/v1");
exports.serviceuser_v1 = v1_66.serviceuser_v1;
var v4_4 = require("./apis/sheets/v4");
exports.sheets_v4 = v4_4.sheets_v4;
var v1_67 = require("./apis/siteVerification/v1");
exports.siteVerification_v1 = v1_67.siteVerification_v1;
var v1_68 = require("./apis/slides/v1");
exports.slides_v1 = v1_68.slides_v1;
var v1_69 = require("./apis/sourcerepo/v1");
exports.sourcerepo_v1 = v1_69.sourcerepo_v1;
var v1_70 = require("./apis/spanner/v1");
exports.spanner_v1 = v1_70.spanner_v1;
var v1explorer_1 = require("./apis/spectrum/v1explorer");
exports.spectrum_v1explorer = v1explorer_1.spectrum_v1explorer;
var v1_71 = require("./apis/speech/v1");
exports.speech_v1 = v1_71.speech_v1;
var v1beta1_15 = require("./apis/speech/v1beta1");
exports.speech_v1beta1 = v1beta1_15.speech_v1beta1;
var v1beta3_2 = require("./apis/sqladmin/v1beta3");
exports.sqladmin_v1beta3 = v1beta3_2.sqladmin_v1beta3;
var v1beta4_2 = require("./apis/sqladmin/v1beta4");
exports.sqladmin_v1beta4 = v1beta4_2.sqladmin_v1beta4;
var v1_72 = require("./apis/storage/v1");
exports.storage_v1 = v1_72.storage_v1;
var v1beta1_16 = require("./apis/storage/v1beta1");
exports.storage_v1beta1 = v1beta1_16.storage_v1beta1;
var v1beta2_7 = require("./apis/storage/v1beta2");
exports.storage_v1beta2 = v1beta2_7.storage_v1beta2;
var v1_73 = require("./apis/storagetransfer/v1");
exports.storagetransfer_v1 = v1_73.storagetransfer_v1;
var v1_74 = require("./apis/streetviewpublish/v1");
exports.streetviewpublish_v1 = v1_74.streetviewpublish_v1;
var v2_20 = require("./apis/surveys/v2");
exports.surveys_v2 = v2_20.surveys_v2;
var v1_75 = require("./apis/tagmanager/v1");
exports.tagmanager_v1 = v1_75.tagmanager_v1;
var v2_21 = require("./apis/tagmanager/v2");
exports.tagmanager_v2 = v2_21.tagmanager_v2;
var v1_76 = require("./apis/tasks/v1");
exports.tasks_v1 = v1_76.tasks_v1;
var v1_77 = require("./apis/testing/v1");
exports.testing_v1 = v1_77.testing_v1;
var v1beta1_17 = require("./apis/texttospeech/v1beta1");
exports.texttospeech_v1beta1 = v1beta1_17.texttospeech_v1beta1;
var v1beta3_3 = require("./apis/toolresults/v1beta3");
exports.toolresults_v1beta3 = v1beta3_3.toolresults_v1beta3;
var v1_78 = require("./apis/tpu/v1");
exports.tpu_v1 = v1_78.tpu_v1;
var v1alpha1_3 = require("./apis/tpu/v1alpha1");
exports.tpu_v1alpha1 = v1alpha1_3.tpu_v1alpha1;
var v2_22 = require("./apis/translate/v2");
exports.translate_v2 = v2_22.translate_v2;
var v1_79 = require("./apis/urlshortener/v1");
exports.urlshortener_v1 = v1_79.urlshortener_v1;
var v1_80 = require("./apis/vault/v1");
exports.vault_v1 = v1_80.vault_v1;
var v1_81 = require("./apis/videointelligence/v1");
exports.videointelligence_v1 = v1_81.videointelligence_v1;
var v1beta2_8 = require("./apis/videointelligence/v1beta2");
exports.videointelligence_v1beta2 = v1beta2_8.videointelligence_v1beta2;
var v1_82 = require("./apis/vision/v1");
exports.vision_v1 = v1_82.vision_v1;
var v1p1beta1_1 = require("./apis/vision/v1p1beta1");
exports.vision_v1p1beta1 = v1p1beta1_1.vision_v1p1beta1;
var v1p2beta1_1 = require("./apis/vision/v1p2beta1");
exports.vision_v1p2beta1 = v1p2beta1_1.vision_v1p2beta1;
var v1_83 = require("./apis/webfonts/v1");
exports.webfonts_v1 = v1_83.webfonts_v1;
var v3_8 = require("./apis/webmasters/v3");
exports.webmasters_v3 = v3_8.webmasters_v3;
var v1alpha_3 = require("./apis/websecurityscanner/v1alpha");
exports.websecurityscanner_v1alpha = v1alpha_3.websecurityscanner_v1alpha;
var v3_9 = require("./apis/youtube/v3");
exports.youtube_v3 = v3_9.youtube_v3;
var v1_84 = require("./apis/youtubeAnalytics/v1");
exports.youtubeAnalytics_v1 = v1_84.youtubeAnalytics_v1;
var v1beta1_18 = require("./apis/youtubeAnalytics/v1beta1");
exports.youtubeAnalytics_v1beta1 = v1beta1_18.youtubeAnalytics_v1beta1;
var v2_23 = require("./apis/youtubeAnalytics/v2");
exports.youtubeAnalytics_v2 = v2_23.youtubeAnalytics_v2;
var v1_85 = require("./apis/youtubereporting/v1");
exports.youtubereporting_v1 = v1_85.youtubereporting_v1;
