// googleapis/build/src/apis/index.js
"use strict";
// Copyright 2012-2016, Google, Inc.
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//    http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (Object.hasOwnProperty.call(mod, k)) result[k] = mod[k];
    result["default"] = mod;
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
const abusiveexperiencereport = __importStar(require("./abusiveexperiencereport"));
const acceleratedmobilepageurl = __importStar(require("./acceleratedmobilepageurl"));
const adexchangebuyer = __importStar(require("./adexchangebuyer"));
const adexchangebuyer2 = __importStar(require("./adexchangebuyer2"));
const adexchangeseller = __importStar(require("./adexchangeseller"));
const adexperiencereport = __importStar(require("./adexperiencereport"));
const admin = __importStar(require("./admin"));
const adsense = __importStar(require("./adsense"));
const adsensehost = __importStar(require("./adsensehost"));
const analytics = __importStar(require("./analytics"));
const analyticsreporting = __importStar(require("./analyticsreporting"));
const androiddeviceprovisioning = __importStar(require("./androiddeviceprovisioning"));
const androidenterprise = __importStar(require("./androidenterprise"));
const androidmanagement = __importStar(require("./androidmanagement"));
const androidpublisher = __importStar(require("./androidpublisher"));
const appengine = __importStar(require("./appengine"));
const appsactivity = __importStar(require("./appsactivity"));
const appstate = __importStar(require("./appstate"));
const bigquery = __importStar(require("./bigquery"));
const bigquerydatatransfer = __importStar(require("./bigquerydatatransfer"));
const blogger = __importStar(require("./blogger"));
const books = __importStar(require("./books"));
const calendar = __importStar(require("./calendar"));
const chat = __importStar(require("./chat"));
const civicinfo = __importStar(require("./civicinfo"));
const classroom = __importStar(require("./classroom"));
const cloudbilling = __importStar(require("./cloudbilling"));
const cloudbuild = __importStar(require("./cloudbuild"));
const clouddebugger = __importStar(require("./clouddebugger"));
const clouderrorreporting = __importStar(require("./clouderrorreporting"));
const cloudfunctions = __importStar(require("./cloudfunctions"));
const cloudiot = __importStar(require("./cloudiot"));
const cloudkms = __importStar(require("./cloudkms"));
const cloudresourcemanager = __importStar(require("./cloudresourcemanager"));
const cloudshell = __importStar(require("./cloudshell"));
const cloudtasks = __importStar(require("./cloudtasks"));
const cloudtrace = __importStar(require("./cloudtrace"));
const composer = __importStar(require("./composer"));
const compute = __importStar(require("./compute"));
const container = __importStar(require("./container"));
const content = __importStar(require("./content"));
const customsearch = __importStar(require("./customsearch"));
const dataflow = __importStar(require("./dataflow"));
const dataproc = __importStar(require("./dataproc"));
const datastore = __importStar(require("./datastore"));
const deploymentmanager = __importStar(require("./deploymentmanager"));
const dfareporting = __importStar(require("./dfareporting"));
const dialogflow = __importStar(require("./dialogflow"));
const digitalassetlinks = __importStar(require("./digitalassetlinks"));
const discovery = __importStar(require("./discovery"));
const dlp = __importStar(require("./dlp"));
const dns = __importStar(require("./dns"));
const doubleclickbidmanager = __importStar(require("./doubleclickbidmanager"));
const doubleclicksearch = __importStar(require("./doubleclicksearch"));
const drive = __importStar(require("./drive"));
const firebasedynamiclinks = __importStar(require("./firebasedynamiclinks"));
const firebaserules = __importStar(require("./firebaserules"));
const firestore = __importStar(require("./firestore"));
const fitness = __importStar(require("./fitness"));
const fusiontables = __importStar(require("./fusiontables"));
const games = __importStar(require("./games"));
const gamesConfiguration = __importStar(require("./gamesConfiguration"));
const gamesManagement = __importStar(require("./gamesManagement"));
const genomics = __importStar(require("./genomics"));
const gmail = __importStar(require("./gmail"));
const groupsmigration = __importStar(require("./groupsmigration"));
const groupssettings = __importStar(require("./groupssettings"));
const iam = __importStar(require("./iam"));
const identitytoolkit = __importStar(require("./identitytoolkit"));
const jobs = __importStar(require("./jobs"));
const kgsearch = __importStar(require("./kgsearch"));
const language = __importStar(require("./language"));
const licensing = __importStar(require("./licensing"));
const logging = __importStar(require("./logging"));
const manufacturers = __importStar(require("./manufacturers"));
const mirror = __importStar(require("./mirror"));
const ml = __importStar(require("./ml"));
const monitoring = __importStar(require("./monitoring"));
const oauth2 = __importStar(require("./oauth2"));
const oslogin = __importStar(require("./oslogin"));
const pagespeedonline = __importStar(require("./pagespeedonline"));
const partners = __importStar(require("./partners"));
const people = __importStar(require("./people"));
const playcustomapp = __importStar(require("./playcustomapp"));
const plus = __importStar(require("./plus"));
const plusDomains = __importStar(require("./plusDomains"));
const poly = __importStar(require("./poly"));
const proximitybeacon = __importStar(require("./proximitybeacon"));
const pubsub = __importStar(require("./pubsub"));
const redis = __importStar(require("./redis"));
const replicapool = __importStar(require("./replicapool"));
const replicapoolupdater = __importStar(require("./replicapoolupdater"));
const reseller = __importStar(require("./reseller"));
const runtimeconfig = __importStar(require("./runtimeconfig"));
const safebrowsing = __importStar(require("./safebrowsing"));
const script = __importStar(require("./script"));
const searchconsole = __importStar(require("./searchconsole"));
const servicebroker = __importStar(require("./servicebroker"));
const serviceconsumermanagement = __importStar(require("./serviceconsumermanagement"));
const servicecontrol = __importStar(require("./servicecontrol"));
const servicemanagement = __importStar(require("./servicemanagement"));
const serviceusage = __importStar(require("./serviceusage"));
const serviceuser = __importStar(require("./serviceuser"));
const sheets = __importStar(require("./sheets"));
const siteVerification = __importStar(require("./siteVerification"));
const slides = __importStar(require("./slides"));
const sourcerepo = __importStar(require("./sourcerepo"));
const spanner = __importStar(require("./spanner"));
const spectrum = __importStar(require("./spectrum"));
const speech = __importStar(require("./speech"));
const sqladmin = __importStar(require("./sqladmin"));
const storage = __importStar(require("./storage"));
const storagetransfer = __importStar(require("./storagetransfer"));
const streetviewpublish = __importStar(require("./streetviewpublish"));
const surveys = __importStar(require("./surveys"));
const tagmanager = __importStar(require("./tagmanager"));
const tasks = __importStar(require("./tasks"));
const testing = __importStar(require("./testing"));
const texttospeech = __importStar(require("./texttospeech"));
const toolresults = __importStar(require("./toolresults"));
const tpu = __importStar(require("./tpu"));
const translate = __importStar(require("./translate"));
const urlshortener = __importStar(require("./urlshortener"));
const vault = __importStar(require("./vault"));
const videointelligence = __importStar(require("./videointelligence"));
const vision = __importStar(require("./vision"));
const webfonts = __importStar(require("./webfonts"));
const webmasters = __importStar(require("./webmasters"));
const websecurityscanner = __importStar(require("./websecurityscanner"));
const youtube = __importStar(require("./youtube"));
const youtubeAnalytics = __importStar(require("./youtubeAnalytics"));
const youtubereporting = __importStar(require("./youtubereporting"));
exports.APIS = {
    abusiveexperiencereport: abusiveexperiencereport.VERSIONS,
    acceleratedmobilepageurl: acceleratedmobilepageurl.VERSIONS,
    adexchangebuyer: adexchangebuyer.VERSIONS,
    adexchangebuyer2: adexchangebuyer2.VERSIONS,
    adexchangeseller: adexchangeseller.VERSIONS,
    adexperiencereport: adexperiencereport.VERSIONS,
    admin: admin.VERSIONS,
    adsense: adsense.VERSIONS,
    adsensehost: adsensehost.VERSIONS,
    analytics: analytics.VERSIONS,
    analyticsreporting: analyticsreporting.VERSIONS,
    androiddeviceprovisioning: androiddeviceprovisioning.VERSIONS,
    androidenterprise: androidenterprise.VERSIONS,
    androidmanagement: androidmanagement.VERSIONS,
    androidpublisher: androidpublisher.VERSIONS,
    appengine: appengine.VERSIONS,
    appsactivity: appsactivity.VERSIONS,
    appstate: appstate.VERSIONS,
    bigquery: bigquery.VERSIONS,
    bigquerydatatransfer: bigquerydatatransfer.VERSIONS,
    blogger: blogger.VERSIONS,
    books: books.VERSIONS,
    calendar: calendar.VERSIONS,
    chat: chat.VERSIONS,
    civicinfo: civicinfo.VERSIONS,
    classroom: classroom.VERSIONS,
    cloudbilling: cloudbilling.VERSIONS,
    cloudbuild: cloudbuild.VERSIONS,
    clouddebugger: clouddebugger.VERSIONS,
    clouderrorreporting: clouderrorreporting.VERSIONS,
    cloudfunctions: cloudfunctions.VERSIONS,
    cloudiot: cloudiot.VERSIONS,
    cloudkms: cloudkms.VERSIONS,
    cloudresourcemanager: cloudresourcemanager.VERSIONS,
    cloudshell: cloudshell.VERSIONS,
    cloudtasks: cloudtasks.VERSIONS,
    cloudtrace: cloudtrace.VERSIONS,
    composer: composer.VERSIONS,
    compute: compute.VERSIONS,
    container: container.VERSIONS,
    content: content.VERSIONS,
    customsearch: customsearch.VERSIONS,
    dataflow: dataflow.VERSIONS,
    dataproc: dataproc.VERSIONS,
    datastore: datastore.VERSIONS,
    deploymentmanager: deploymentmanager.VERSIONS,
    dfareporting: dfareporting.VERSIONS,
    dialogflow: dialogflow.VERSIONS,
    digitalassetlinks: digitalassetlinks.VERSIONS,
    discovery: discovery.VERSIONS,
    dlp: dlp.VERSIONS,
    dns: dns.VERSIONS,
    doubleclickbidmanager: doubleclickbidmanager.VERSIONS,
    doubleclicksearch: doubleclicksearch.VERSIONS,
    drive: drive.VERSIONS,
    firebasedynamiclinks: firebasedynamiclinks.VERSIONS,
    firebaserules: firebaserules.VERSIONS,
    firestore: firestore.VERSIONS,
    fitness: fitness.VERSIONS,
    fusiontables: fusiontables.VERSIONS,
    games: games.VERSIONS,
    gamesConfiguration: gamesConfiguration.VERSIONS,
    gamesManagement: gamesManagement.VERSIONS,
    genomics: genomics.VERSIONS,
    gmail: gmail.VERSIONS,
    groupsmigration: groupsmigration.VERSIONS,
    groupssettings: groupssettings.VERSIONS,
    iam: iam.VERSIONS,
    identitytoolkit: identitytoolkit.VERSIONS,
    jobs: jobs.VERSIONS,
    kgsearch: kgsearch.VERSIONS,
    language: language.VERSIONS,
    licensing: licensing.VERSIONS,
    logging: logging.VERSIONS,
    manufacturers: manufacturers.VERSIONS,
    mirror: mirror.VERSIONS,
    ml: ml.VERSIONS,
    monitoring: monitoring.VERSIONS,
    oauth2: oauth2.VERSIONS,
    oslogin: oslogin.VERSIONS,
    pagespeedonline: pagespeedonline.VERSIONS,
    partners: partners.VERSIONS,
    people: people.VERSIONS,
    playcustomapp: playcustomapp.VERSIONS,
    plus: plus.VERSIONS,
    plusDomains: plusDomains.VERSIONS,
    poly: poly.VERSIONS,
    proximitybeacon: proximitybeacon.VERSIONS,
    pubsub: pubsub.VERSIONS,
    redis: redis.VERSIONS,
    replicapool: replicapool.VERSIONS,
    replicapoolupdater: replicapoolupdater.VERSIONS,
    reseller: reseller.VERSIONS,
    runtimeconfig: runtimeconfig.VERSIONS,
    safebrowsing: safebrowsing.VERSIONS,
    script: script.VERSIONS,
    searchconsole: searchconsole.VERSIONS,
    servicebroker: servicebroker.VERSIONS,
    serviceconsumermanagement: serviceconsumermanagement.VERSIONS,
    servicecontrol: servicecontrol.VERSIONS,
    servicemanagement: servicemanagement.VERSIONS,
    serviceusage: serviceusage.VERSIONS,
    serviceuser: serviceuser.VERSIONS,
    sheets: sheets.VERSIONS,
    siteVerification: siteVerification.VERSIONS,
    slides: slides.VERSIONS,
    sourcerepo: sourcerepo.VERSIONS,
    spanner: spanner.VERSIONS,
    spectrum: spectrum.VERSIONS,
    speech: speech.VERSIONS,
    sqladmin: sqladmin.VERSIONS,
    storage: storage.VERSIONS,
    storagetransfer: storagetransfer.VERSIONS,
    streetviewpublish: streetviewpublish.VERSIONS,
    surveys: surveys.VERSIONS,
    tagmanager: tagmanager.VERSIONS,
    tasks: tasks.VERSIONS,
    testing: testing.VERSIONS,
    texttospeech: texttospeech.VERSIONS,
    toolresults: toolresults.VERSIONS,
    tpu: tpu.VERSIONS,
    translate: translate.VERSIONS,
    urlshortener: urlshortener.VERSIONS,
    vault: vault.VERSIONS,
    videointelligence: videointelligence.VERSIONS,
    vision: vision.VERSIONS,
    webfonts: webfonts.VERSIONS,
    webmasters: webmasters.VERSIONS,
    websecurityscanner: websecurityscanner.VERSIONS,
    youtube: youtube.VERSIONS,
    youtubeAnalytics: youtubeAnalytics.VERSIONS,
    youtubereporting: youtubereporting.VERSIONS,
};
class GeneratedAPIs {
    constructor() {
        this.abusiveexperiencereport = abusiveexperiencereport.abusiveexperiencereport;
        this.acceleratedmobilepageurl = acceleratedmobilepageurl.acceleratedmobilepageurl;
        this.adexchangebuyer = adexchangebuyer.adexchangebuyer;
        this.adexchangebuyer2 = adexchangebuyer2.adexchangebuyer2;
        this.adexchangeseller = adexchangeseller.adexchangeseller;
        this.adexperiencereport = adexperiencereport.adexperiencereport;
        this.admin = admin.admin;
        this.adsense = adsense.adsense;
        this.adsensehost = adsensehost.adsensehost;
        this.analytics = analytics.analytics;
        this.analyticsreporting = analyticsreporting.analyticsreporting;
        this.androiddeviceprovisioning = androiddeviceprovisioning.androiddeviceprovisioning;
        this.androidenterprise = androidenterprise.androidenterprise;
        this.androidmanagement = androidmanagement.androidmanagement;
        this.androidpublisher = androidpublisher.androidpublisher;
        this.appengine = appengine.appengine;
        this.appsactivity = appsactivity.appsactivity;
        this.appstate = appstate.appstate;
        this.bigquery = bigquery.bigquery;
        this.bigquerydatatransfer = bigquerydatatransfer.bigquerydatatransfer;
        this.blogger = blogger.blogger;
        this.books = books.books;
        this.calendar = calendar.calendar;
        this.chat = chat.chat;
        this.civicinfo = civicinfo.civicinfo;
        this.classroom = classroom.classroom;
        this.cloudbilling = cloudbilling.cloudbilling;
        this.cloudbuild = cloudbuild.cloudbuild;
        this.clouddebugger = clouddebugger.clouddebugger;
        this.clouderrorreporting = clouderrorreporting.clouderrorreporting;
        this.cloudfunctions = cloudfunctions.cloudfunctions;
        this.cloudiot = cloudiot.cloudiot;
        this.cloudkms = cloudkms.cloudkms;
        this.cloudresourcemanager = cloudresourcemanager.cloudresourcemanager;
        this.cloudshell = cloudshell.cloudshell;
        this.cloudtasks = cloudtasks.cloudtasks;
        this.cloudtrace = cloudtrace.cloudtrace;
        this.composer = composer.composer;
        this.compute = compute.compute;
        this.container = container.container;
        this.content = content.content;
        this.customsearch = customsearch.customsearch;
        this.dataflow = dataflow.dataflow;
        this.dataproc = dataproc.dataproc;
        this.datastore = datastore.datastore;
        this.deploymentmanager = deploymentmanager.deploymentmanager;
        this.dfareporting = dfareporting.dfareporting;
        this.dialogflow = dialogflow.dialogflow;
        this.digitalassetlinks = digitalassetlinks.digitalassetlinks;
        this.discovery = discovery.discovery;
        this.dlp = dlp.dlp;
        this.dns = dns.dns;
        this.doubleclickbidmanager = doubleclickbidmanager.doubleclickbidmanager;
        this.doubleclicksearch = doubleclicksearch.doubleclicksearch;
        this.drive = drive.drive;
        this.firebasedynamiclinks = firebasedynamiclinks.firebasedynamiclinks;
        this.firebaserules = firebaserules.firebaserules;
        this.firestore = firestore.firestore;
        this.fitness = fitness.fitness;
        this.fusiontables = fusiontables.fusiontables;
        this.games = games.games;
        this.gamesConfiguration = gamesConfiguration.gamesConfiguration;
        this.gamesManagement = gamesManagement.gamesManagement;
        this.genomics = genomics.genomics;
        this.gmail = gmail.gmail;
        this.groupsmigration = groupsmigration.groupsmigration;
        this.groupssettings = groupssettings.groupssettings;
        this.iam = iam.iam;
        this.identitytoolkit = identitytoolkit.identitytoolkit;
        this.jobs = jobs.jobs;
        this.kgsearch = kgsearch.kgsearch;
        this.language = language.language;
        this.licensing = licensing.licensing;
        this.logging = logging.logging;
        this.manufacturers = manufacturers.manufacturers;
        this.mirror = mirror.mirror;
        this.ml = ml.ml;
        this.monitoring = monitoring.monitoring;
        this.oauth2 = oauth2.oauth2;
        this.oslogin = oslogin.oslogin;
        this.pagespeedonline = pagespeedonline.pagespeedonline;
        this.partners = partners.partners;
        this.people = people.people;
        this.playcustomapp = playcustomapp.playcustomapp;
        this.plus = plus.plus;
        this.plusDomains = plusDomains.plusDomains;
        this.poly = poly.poly;
        this.proximitybeacon = proximitybeacon.proximitybeacon;
        this.pubsub = pubsub.pubsub;
        this.redis = redis.redis;
        this.replicapool = replicapool.replicapool;
        this.replicapoolupdater = replicapoolupdater.replicapoolupdater;
        this.reseller = reseller.reseller;
        this.runtimeconfig = runtimeconfig.runtimeconfig;
        this.safebrowsing = safebrowsing.safebrowsing;
        this.script = script.script;
        this.searchconsole = searchconsole.searchconsole;
        this.servicebroker = servicebroker.servicebroker;
        this.serviceconsumermanagement = serviceconsumermanagement.serviceconsumermanagement;
        this.servicecontrol = servicecontrol.servicecontrol;
        this.servicemanagement = servicemanagement.servicemanagement;
        this.serviceusage = serviceusage.serviceusage;
        this.serviceuser = serviceuser.serviceuser;
        this.sheets = sheets.sheets;
        this.siteVerification = siteVerification.siteVerification;
        this.slides = slides.slides;
        this.sourcerepo = sourcerepo.sourcerepo;
        this.spanner = spanner.spanner;
        this.spectrum = spectrum.spectrum;
        this.speech = speech.speech;
        this.sqladmin = sqladmin.sqladmin;
        this.storage = storage.storage;
        this.storagetransfer = storagetransfer.storagetransfer;
        this.streetviewpublish = streetviewpublish.streetviewpublish;
        this.surveys = surveys.surveys;
        this.tagmanager = tagmanager.tagmanager;
        this.tasks = tasks.tasks;
        this.testing = testing.testing;
        this.texttospeech = texttospeech.texttospeech;
        this.toolresults = toolresults.toolresults;
        this.tpu = tpu.tpu;
        this.translate = translate.translate;
        this.urlshortener = urlshortener.urlshortener;
        this.vault = vault.vault;
        this.videointelligence = videointelligence.videointelligence;
        this.vision = vision.vision;
        this.webfonts = webfonts.webfonts;
        this.webmasters = webmasters.webmasters;
        this.websecurityscanner = websecurityscanner.websecurityscanner;
        this.youtube = youtube.youtube;
        this.youtubeAnalytics = youtubeAnalytics.youtubeAnalytics;
        this.youtubereporting = youtubereporting.youtubereporting;
        this.abusiveexperiencereport = this.abusiveexperiencereport.bind(this);
        this.acceleratedmobilepageurl = this.acceleratedmobilepageurl.bind(this);
        this.adexchangebuyer = this.adexchangebuyer.bind(this);
        this.adexchangebuyer2 = this.adexchangebuyer2.bind(this);
        this.adexchangeseller = this.adexchangeseller.bind(this);
        this.adexperiencereport = this.adexperiencereport.bind(this);
        this.admin = this.admin.bind(this);
        this.adsense = this.adsense.bind(this);
        this.adsensehost = this.adsensehost.bind(this);
        this.analytics = this.analytics.bind(this);
        this.analyticsreporting = this.analyticsreporting.bind(this);
        this.androiddeviceprovisioning = this.androiddeviceprovisioning.bind(this);
        this.androidenterprise = this.androidenterprise.bind(this);
        this.androidmanagement = this.androidmanagement.bind(this);
        this.androidpublisher = this.androidpublisher.bind(this);
        this.appengine = this.appengine.bind(this);
        this.appsactivity = this.appsactivity.bind(this);
        this.appstate = this.appstate.bind(this);
        this.bigquery = this.bigquery.bind(this);
        this.bigquerydatatransfer = this.bigquerydatatransfer.bind(this);
        this.blogger = this.blogger.bind(this);
        this.books = this.books.bind(this);
        this.calendar = this.calendar.bind(this);
        this.chat = this.chat.bind(this);
        this.civicinfo = this.civicinfo.bind(this);
        this.classroom = this.classroom.bind(this);
        this.cloudbilling = this.cloudbilling.bind(this);
        this.cloudbuild = this.cloudbuild.bind(this);
        this.clouddebugger = this.clouddebugger.bind(this);
        this.clouderrorreporting = this.clouderrorreporting.bind(this);
        this.cloudfunctions = this.cloudfunctions.bind(this);
        this.cloudiot = this.cloudiot.bind(this);
        this.cloudkms = this.cloudkms.bind(this);
        this.cloudresourcemanager = this.cloudresourcemanager.bind(this);
        this.cloudshell = this.cloudshell.bind(this);
        this.cloudtasks = this.cloudtasks.bind(this);
        this.cloudtrace = this.cloudtrace.bind(this);
        this.composer = this.composer.bind(this);
        this.compute = this.compute.bind(this);
        this.container = this.container.bind(this);
        this.content = this.content.bind(this);
        this.customsearch = this.customsearch.bind(this);
        this.dataflow = this.dataflow.bind(this);
        this.dataproc = this.dataproc.bind(this);
        this.datastore = this.datastore.bind(this);
        this.deploymentmanager = this.deploymentmanager.bind(this);
        this.dfareporting = this.dfareporting.bind(this);
        this.dialogflow = this.dialogflow.bind(this);
        this.digitalassetlinks = this.digitalassetlinks.bind(this);
        this.discovery = this.discovery.bind(this);
        this.dlp = this.dlp.bind(this);
        this.dns = this.dns.bind(this);
        this.doubleclickbidmanager = this.doubleclickbidmanager.bind(this);
        this.doubleclicksearch = this.doubleclicksearch.bind(this);
        this.drive = this.drive.bind(this);
        this.firebasedynamiclinks = this.firebasedynamiclinks.bind(this);
        this.firebaserules = this.firebaserules.bind(this);
        this.firestore = this.firestore.bind(this);
        this.fitness = this.fitness.bind(this);
        this.fusiontables = this.fusiontables.bind(this);
        this.games = this.games.bind(this);
        this.gamesConfiguration = this.gamesConfiguration.bind(this);
        this.gamesManagement = this.gamesManagement.bind(this);
        this.genomics = this.genomics.bind(this);
        this.gmail = this.gmail.bind(this);
        this.groupsmigration = this.groupsmigration.bind(this);
        this.groupssettings = this.groupssettings.bind(this);
        this.iam = this.iam.bind(this);
        this.identitytoolkit = this.identitytoolkit.bind(this);
        this.jobs = this.jobs.bind(this);
        this.kgsearch = this.kgsearch.bind(this);
        this.language = this.language.bind(this);
        this.licensing = this.licensing.bind(this);
        this.logging = this.logging.bind(this);
        this.manufacturers = this.manufacturers.bind(this);
        this.mirror = this.mirror.bind(this);
        this.ml = this.ml.bind(this);
        this.monitoring = this.monitoring.bind(this);
        this.oauth2 = this.oauth2.bind(this);
        this.oslogin = this.oslogin.bind(this);
        this.pagespeedonline = this.pagespeedonline.bind(this);
        this.partners = this.partners.bind(this);
        this.people = this.people.bind(this);
        this.playcustomapp = this.playcustomapp.bind(this);
        this.plus = this.plus.bind(this);
        this.plusDomains = this.plusDomains.bind(this);
        this.poly = this.poly.bind(this);
        this.proximitybeacon = this.proximitybeacon.bind(this);
        this.pubsub = this.pubsub.bind(this);
        this.redis = this.redis.bind(this);
        this.replicapool = this.replicapool.bind(this);
        this.replicapoolupdater = this.replicapoolupdater.bind(this);
        this.reseller = this.reseller.bind(this);
        this.runtimeconfig = this.runtimeconfig.bind(this);
        this.safebrowsing = this.safebrowsing.bind(this);
        this.script = this.script.bind(this);
        this.searchconsole = this.searchconsole.bind(this);
        this.servicebroker = this.servicebroker.bind(this);
        this.serviceconsumermanagement = this.serviceconsumermanagement.bind(this);
        this.servicecontrol = this.servicecontrol.bind(this);
        this.servicemanagement = this.servicemanagement.bind(this);
        this.serviceusage = this.serviceusage.bind(this);
        this.serviceuser = this.serviceuser.bind(this);
        this.sheets = this.sheets.bind(this);
        this.siteVerification = this.siteVerification.bind(this);
        this.slides = this.slides.bind(this);
        this.sourcerepo = this.sourcerepo.bind(this);
        this.spanner = this.spanner.bind(this);
        this.spectrum = this.spectrum.bind(this);
        this.speech = this.speech.bind(this);
        this.sqladmin = this.sqladmin.bind(this);
        this.storage = this.storage.bind(this);
        this.storagetransfer = this.storagetransfer.bind(this);
        this.streetviewpublish = this.streetviewpublish.bind(this);
        this.surveys = this.surveys.bind(this);
        this.tagmanager = this.tagmanager.bind(this);
        this.tasks = this.tasks.bind(this);
        this.testing = this.testing.bind(this);
        this.texttospeech = this.texttospeech.bind(this);
        this.toolresults = this.toolresults.bind(this);
        this.tpu = this.tpu.bind(this);
        this.translate = this.translate.bind(this);
        this.urlshortener = this.urlshortener.bind(this);
        this.vault = this.vault.bind(this);
        this.videointelligence = this.videointelligence.bind(this);
        this.vision = this.vision.bind(this);
        this.webfonts = this.webfonts.bind(this);
        this.webmasters = this.webmasters.bind(this);
        this.websecurityscanner = this.websecurityscanner.bind(this);
        this.youtube = this.youtube.bind(this);
        this.youtubeAnalytics = this.youtubeAnalytics.bind(this);
        this.youtubereporting = this.youtubereporting.bind(this);
    }
}
exports.GeneratedAPIs = GeneratedAPIs;
